index.d.ts

```diff
@@ -0,0 +1,758 @@
+declare module 'extra-array' {
+  /**
+   * Binary searches leftmost value in sorted array.
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} first index of value | ~(index of closest value)
+   */
+  export function bsearch<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
```

Comment:
```suggestion
  export function bsearch<T>(x: T[], v: T, fn?: (a: T, b: T) => number): number;
```

Reply:
`compare()` function returns a number like: x<y: -1, x=y: 0, x>y: 1
This is similar to Array.sort()

Reply:
This and other comments about compare function noted, added a compareFn type :+1: